src/index.js

```diff
@@ -206,12 +205,11 @@ class Service {
   update(id, data) {
     return this._get(id).then(getData => {
       data.id = id;
-
       return this.table.get(getData.id)
         .replace(data, {
           returnChanges: true
         }).run()
-        .then(result => result.changes[0].new_val);
+        .then(result => (result.changes && result.changes.length) ? result.changes[0].new_val : {});
```

Comment:
ok looks like @daffl caught this: https://github.com/feathersjs/feathers-rethinkdb/blob/master/src/index.js#L213.